WaterLog_Backend/Controllers/SegmentEventsController.cs

```diff
@@ -190,290 +197,55 @@ public async Task<List<GetAlerts>> GetAlertsByPage(int id = 1)
 
-            List<GetAlerts> ListOfAlerts = new List<GetAlerts>();
-            //Get Entries from SegmentLeaks
-            //filter for segmentId
-            if (segment != 0)
+            if (filter.SensorType != 0)
             {
-                if (severity != 0)
+                var sensorData = new List<GetAlerts>();
+                if (filter.SensorType == 1)
                 {
-                    var leaksQuery = await _db.SegmentLeaks
-                                          .Where(a => a.SegmentsId == segment && a.Severity == tableSeverity)
-                                          .OrderByDescending(a => a.OriginalTimeStamp)
-                                          .OrderByDescending(b => b.ResolvedStatus).Skip((1 - 1) * Globals.NumberItems)
-                                          .Take(Globals.NumberItems).ToListAsync();
-                    if (leaksQuery.Count != 0)
-                    {
-                        var proc = new Procedures(_db, _config);
-                        foreach (SegmentLeaksEntry entry in leaksQuery)
-                        {
-                            double totalSystemLitres = await proc.CalculateTotalUsageLitres(entry),
-                                    litresUsed = await proc.CalculateTotalWastageLitres(entry),
-                                    perhourwastagelitre = await proc.CalculatePerHourWastageLitre(entry),
-                                    cost = await proc.CalculatePerHourWastageCost(entry);
-
-                            //Find Litre Usage
-                            ListOfAlerts.Add
-                             (
-                                 new GetAlerts
-                                 (
-                                     entry.OriginalTimeStamp,
-                                     (entry
-                                     .LatestTimeStamp
-                                     .Subtract(entry
-                                     .OriginalTimeStamp) < TimeSpan.Zero ? TimeSpan.Zero : entry
-                                     .LatestTimeStamp.
-                                     Subtract(entry
-                                     .OriginalTimeStamp)),
-                                     Globals.Segment,
-                                     entry.SegmentsId,
-                                     Globals.Leak,
-                                     cost,
-                                     perhourwastagelitre,
-                                     entry.Severity,
-                                     litresUsed,
-                                     totalSystemLitres,
-                                     entry.ResolvedStatus
-                                  )
-                             );
-                        }
-                    }
+                    onlySeverity = false;
+                    sensorData = alerts
+                        .Where(alert => alert.EntityName.ToLower() == "tank" && alert.EntityId == filter.SensorId)
```

Comment:
add entity name to an enum instead of giving string value here to "tank" and "sensor"

Reply:
replaced string with entity name